src/lib.rs

```diff
@@ -320,6 +320,224 @@ impl DltExtendedHeader {
             self.message_info &= 0b1111_1110;
         }
     }
+
+    ///Returns message type or `Option::None` for reserved values.
+    pub fn mstp(&self) -> Option<Mstp> {
+        Mstp::from_u8((self.message_info >> 1) & 0b111)
+    }
+
+    ///Returns message type info or `Option::None` for reserved values.
+    pub fn mtin(&self) -> Option<Mtin> {
+        let mstp = self.mstp()?;
+        Mtin::from_u8(mstp, (self.message_info >> 4) & 0b1111)
+    }
+
+    ///Set message type info and based on that the message type.
+    pub fn set_mstp_mtin(&mut self, mtin: Mtin) {
+        match mtin {
+            Mtin::DltLogMessage(mtin) => {
+                self.message_info |= ((Mstp::DltTypeLog as u8) << 1) & 0b0000_1110;
+                self.message_info |= ((mtin as u8) << 4) & 0b1111_0000;
+            },
+            Mtin::DltTraceMessage(mtin) => {
+                self.message_info |= ((Mstp::DltTypeAppTrace as u8)<< 1) & 0b0000_1110;
+                self.message_info |= ((mtin as u8) << 4) & 0b1111_0000;
+            },
+            Mtin::DltNetworkMessage(mtin) => {
+                self.message_info |= ((Mstp::DltTypeNwTrace as u8) << 1) & 0b0000_1110;
+                self.message_info |= (mtin.to_u8() << 4) & 0b1111_0000;
+            },
+            Mtin::DltControlMessage(mtin) => {
+                self.message_info |= ((Mstp::DltTypeControl as u8) << 1) & 0b0000_1110;
+                self.message_info |= ((mtin as u8) << 4) & 0b1111_0000;
+            },
+        }
+    }
+}
+
+///Message type
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Mstp {
+    ///Dlt log message.
+    DltTypeLog = 0x0,
+    ///Dlt trace message.
+    DltTypeAppTrace = 0x1,
+    ///Dlt network message.
+    DltTypeNwTrace = 0x2,
+    ///Dlt control message.
+    DltTypeControl = 0x3
+}
+
+impl Mstp {
+    ///Attempts to convert a raw mstp value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mstp> {
+        match value {
+            0x0 => Some(Mstp::DltTypeLog),
+            0x1 => Some(Mstp::DltTypeAppTrace),
+            0x2 => Some(Mstp::DltTypeNwTrace),
+            0x3 => Some(Mstp::DltTypeControl),
+            //Reserved values
+            _ => None,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MtinDltLogMessage {
+    ///Fatal system error.
+    DltLogFatal = 0x1,
+    ///SWC error.
+    DltLogError = 0x2,
+    ///Correct behavior cannot be ensured.
+    DltLogWarn = 0x3,
+    ///Message of LogLevel type “Information”.
+    DltLogInfo = 0x4,
+    ///Message of LogLevel type “Debug”.
+    DltLogDebug = 0x5,
+    ///Message of LogLevel type "Verbose".
+    DltLogVerbose = 0x6,
+}
+
+impl MtinDltLogMessage {
+    ///Attempts to convert a raw mtin value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mtin> {
+        match value {
```

Comment:
The spec does not describe the meaning of 0x0 for mtins so I am assuming this is reserved for future use as well or 0x0 means this is optional.

Reply:
Makes sense